src/main/java/com/marklogic/spark/reader/ReadContext.java

```diff
@@ -69,7 +69,9 @@ public ReadContext(Map<String, String> properties, boolean inferSchema) {
             SparkSession.active().sparkContext().defaultMinPartitions(), 1);
         final long batchSize = getNumericOption(ReadConstants.BATCH_SIZE, DEFAULT_BATCH_SIZE, 0);
         final String dslQuery = properties.get(ReadConstants.OPTIC_DSL);
-
+        if (dslQuery == null || dslQuery.trim().length() < 1) {
```

Comment:
Added this as I forgot the query in one of my ad hoc tests, and the error didn't indicate what the problem was